diff --git a/Makefile b/Makefile
index 8f4a6f8..d08b59a 100644
--- a/Makefile
+++ b/Makefile
@@ -108,6 +108,9 @@ endif
 
 ifeq ($(GPU_SUPPORT),true)
 	override ADDFLAGS += -DGPU_SUPPORT
+	ifeq ($(PLATFORM_LC),linux)
+		override LDFLAGS += -ldl
+	endif
 endif
 
 #? Compiler and Linker
diff --git a/src/linux/intel_gpu_top/intel_gpu_top.c b/src/linux/intel_gpu_top/intel_gpu_top.c
index e466c5f..71419f7 100644
--- a/src/linux/intel_gpu_top/intel_gpu_top.c
+++ b/src/linux/intel_gpu_top/intel_gpu_top.c
@@ -21,6 +21,10 @@
  * DEALINGS IN THE SOFTWARE.
  */
 
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+
 #include <assert.h>
 #include <ctype.h>
 #include <dirent.h>
